lib/rex/socket/comm/local.rb

```diff
@@ -194,15 +198,21 @@ def self.create_by_type(param, type, proto = 0)
       sock.setsockopt(::Socket::SOL_SOCKET, ::Socket::SO_BROADCAST, true)
     end
 
-    # If a server TCP instance is being created...
+    # If a server instance is being created...
     if param.server?
       sock.listen(256)
 
       if !param.bare?
-        klass = Rex::Socket::TcpServer
-        if param.ssl
-          klass = Rex::Socket::SslTcpServer
-        end
+        if param.proto == 'tcp'
+          klass = Rex::Socket::TcpServer
+          if param.ssl
+            klass = Rex::Socket::SslTcpServer
+          end
+        elsif param.proto == 'sctp'
+            klass = Rex::Socket::SctpServer
+        else
+          raise Rex::BindFailed.new(param.localhost, param.localport), caller
```

Comment:
not a blocker: I think this is the right exception to raise here, but I worry it might be hard to debug if this code path ever triggers as it's probably caught and silently ignored in some cases in framework

Reply:
If that is the case, (IMO) we should fix it higher up in the stack where it is erroneously ignored.
I look at `Rex` as the "kernel" of Metasploit which needs to be consistent and informative as much as possible, especially going down to the core (Socket, IO, Event, locking, all that fun junk). That said, `Error`s are `Object`s too and we can do whatever we want with them - like mixing in/extending with an ancestor which will increase chances of trapping on `Socket` errors...